### PR TITLE
Replace prim__getArgs with prim__getArgsCount and prim__getArg

### PR DESCRIPTION
**Pull request by melted**
This spares us from having to build Idris data structures in the foreign
code.

### CODE REVIEW

**Comment by melted** [comment]
I don't understand the CI failure at all.

**Comment by gallais** [comment]
The problem happens earlier already cf. build-bootstrap-racket:
https://github.com/idris-lang/Idris2/pull/1269/checks?check_run_id=2288909053#step:4:753

I'm not sure why though.

File: support/racket/support.rkt [hunk 1]
```diff
@@ -409,14 +409,13 @@
...
 (define (blodwen-clock-nanosecond time) (time-nanosecond time))
 
-(define (blodwen-args)
-  (define (blodwen-build-args args)
-    (if (null? args)
-        (vector 0) ; Prelude.List
-        (vector 1 (car args) (blodwen-build-args (cdr args)))))
-    (blodwen-build-args
-      (cons (path->string (find-system-path 'run-file))
-            (vector->list (current-command-line-arguments)))))
```
Comment (gallais) [inline support/racket/support.rkt:419]: By using `current-command-line-arguments` you accidentally threw away
`path->string (find-system-path 'run-file)` as the first argument.

I expect that somewhere Idris assumes it gets the name of the executable
first & then the args and that breaks that assumption?

You should probably have something like:

```racket
(define (blodwen-arg-count)
  (+ 1 (vector-length (current-command-line-arguments))))

(define (blodwen-arg n)
  (cond 
     [(= 0 n) (path->string (find-system-path 'run-file))]
     [(< (- n 1) (vector-length (current-command-line-arguments)))
         (vector-ref (current-command-line-arguments) (- n 1))]
     [else ""])
```

**Review by gallais — CHANGES_REQUESTED** [review]
May have found the bug

File: support/racket/support.rkt (hunk 1 above)
Comment (melted) [reply support/racket/support.rkt:419]: Ok!

**Comment by melted** [comment]
Ooops.

**Comment by melted** [comment]
Parentheses are hard. This should work.

**Pull request merged**